(azureml_py310_sdkv2) azureuser@shekharh1002025:/mnt/batch/tasks/shared/LS_root/mounts/clusters/shekharh1002025/code/Users/shekhar_icd/src/model$ torchrun --nproc_per_node=2 optimized_training_script_v4.py
W0813 16:50:01.204000 13178 site-packages/torch/distributed/run.py:766] 
W0813 16:50:01.204000 13178 site-packages/torch/distributed/run.py:766] *****************************************
W0813 16:50:01.204000 13178 site-packages/torch/distributed/run.py:766] Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
W0813 16:50:01.204000 13178 site-packages/torch/distributed/run.py:766] *****************************************
[W813 16:50:01.571463581 socket.cpp:200] [c10d] The hostname of the client socket cannot be retrieved. err=-3
[W813 16:50:01.574643143 socket.cpp:200] [c10d] The hostname of the client socket cannot be retrieved. err=-3
Load Parameters
Load Parameters
Loading Datasets
Loading Datasets
Loading dataset shards: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 18/18 [00:00<00:00, 31.43it/s]
Loading dataset shards: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 18/18 [00:00<00:00, 21.03it/s]
Setting Up Dataloader Num Workers & PIN Meory Flag
Recommended dataloader_num_workers : 6
AVAILABLE_MEMORY_GB : 620.5052680969238 | DATASET_MEMORY_SIZE_GB : 8.103603198051452 pin_memory : True
Datasets Loaded
..............
Loading Model e5-large-v2
Setting Up Dataloader Num Workers & PIN Meory Flag
Recommended dataloader_num_workers : 6
AVAILABLE_MEMORY_GB : 620.4955978393555 | DATASET_MEMORY_SIZE_GB : 8.103603198051452 pin_memory : True
Datasets Loaded
..............
Loading Model e5-large-v2
e5-large-v2 Model Initialized
...........
Initalizing Loss : MNR
Loss Initialized
..............
Initializing Optimizer
Learning Rate Scheduler Selected : cosine
Load Sentence Transformer Training Arguments
e5-large-v2 Model Initialized
...........
Initalizing Loss : MNR
Loss Initialized
..............
Initializing Optimizer
Learning Rate Scheduler Selected : cosine
Load Sentence Transformer Training Arguments
[W813 16:50:58.940182427 socket.cpp:200] [c10d] The hostname of the client socket cannot be retrieved. err=-3
[W813 16:50:58.941789674 socket.cpp:200] [c10d] The hostname of the client socket cannot be retrieved. err=-3
[W813 16:50:58.943095407 socket.cpp:200] [c10d] The hostname of the client socket cannot be retrieved. err=-3
[W813 16:50:58.945990202 socket.cpp:200] [c10d] The hostname of the client socket cannot be retrieved. err=-3
/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azureml/core/__init__.py:11: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.
  import pkg_resources
/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azureml/core/__init__.py:11: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.
  import pkg_resources
Training Started                                                                                                                                                                                              
Training Started                                                                                                                                                                                              
  0%|                                                                                                                                                                              | 0/124800 [00:00<?, ?it/s][rank1]: Traceback (most recent call last):
[rank1]:   File "/mnt/batch/tasks/shared/LS_root/mounts/clusters/shekharh1002025/code/Users/shekhar_icd/src/model/optimized_training_script_v4.py", line 489, in <module>
[rank1]:     main(config, date_time)
[rank1]:   File "/mnt/batch/tasks/shared/LS_root/mounts/clusters/shekharh1002025/code/Users/shekhar_icd/src/model/optimized_training_script_v4.py", line 466, in main
[rank1]:     trainer.train()
[rank1]:   File "/home/azureuser/.local/lib/python3.10/site-packages/transformers/trainer.py", line 2206, in train
[rank1]:     return inner_training_loop(
[rank1]:   File "/home/azureuser/.local/lib/python3.10/site-packages/transformers/trainer.py", line 2548, in _inner_training_loop
[rank1]:     tr_loss_step = self.training_step(model, inputs, num_items_in_batch)
[rank1]:   File "/home/azureuser/.local/lib/python3.10/site-packages/transformers/trainer.py", line 3749, in training_step
[rank1]:     loss = self.compute_loss(model, inputs, num_items_in_batch=num_items_in_batch)
[rank1]:   File "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/sentence_transformers/trainer.py", line 406, in compute_loss
[rank1]:     loss = loss_fn(features, labels)
[rank1]:   File "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
[rank1]:     return self._call_impl(*args, **kwargs)
[rank1]:   File "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
[rank1]:     return forward_call(*args, **kwargs)
[rank1]:   File "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/sentence_transformers/losses/MultipleNegativesRankingLoss.py", line 103, in forward
[rank1]:     anchors = embeddings[0]  # (batch_size, embedding_dim)
[rank1]: IndexError: list index out of range
[rank0]: Traceback (most recent call last):
[rank0]:   File "/mnt/batch/tasks/shared/LS_root/mounts/clusters/shekharh1002025/code/Users/shekhar_icd/src/model/optimized_training_script_v4.py", line 489, in <module>
[rank0]:     main(config, date_time)
[rank0]:   File "/mnt/batch/tasks/shared/LS_root/mounts/clusters/shekharh1002025/code/Users/shekhar_icd/src/model/optimized_training_script_v4.py", line 466, in main
[rank0]:     trainer.train()
[rank0]:   File "/home/azureuser/.local/lib/python3.10/site-packages/transformers/trainer.py", line 2206, in train
[rank0]:     return inner_training_loop(
[rank0]:   File "/home/azureuser/.local/lib/python3.10/site-packages/transformers/trainer.py", line 2548, in _inner_training_loop
[rank0]:     tr_loss_step = self.training_step(model, inputs, num_items_in_batch)
[rank0]:   File "/home/azureuser/.local/lib/python3.10/site-packages/transformers/trainer.py", line 3749, in training_step
[rank0]:     loss = self.compute_loss(model, inputs, num_items_in_batch=num_items_in_batch)
[rank0]:   File "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/sentence_transformers/trainer.py", line 406, in compute_loss
[rank0]:     loss = loss_fn(features, labels)
[rank0]:   File "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
[rank0]:     return self._call_impl(*args, **kwargs)
[rank0]:   File "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
[rank0]:     return forward_call(*args, **kwargs)
[rank0]:   File "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/sentence_transformers/losses/MultipleNegativesRankingLoss.py", line 103, in forward
[rank0]:     anchors = embeddings[0]  # (batch_size, embedding_dim)
[rank0]: IndexError: list index out of range

